cmd/bbolt/main.go

```diff
@@ -1828,14 +1837,21 @@ func ReadPageSize(path string) (int, error) {
 
 	// Read page size from metadata.
 	m := (*meta)(unsafe.Pointer(&buf[PageHeaderSize]))
-	return int(m.pageSize), nil
+	if m.magic != bolt.Magic {
+		return 0, fmt.Errorf("the page has wrong (unexpected) magic")
+	}
```

Comment:
Since the command is supposed to be executed by users. They need to use `check` command to validate the DB. In other words, even the db is corrupted, they should also run some command, such as dumping page items. 

```suggestion
```

Reply:
In https://github.com/etcd-io/bbolt/pull/354. 

We need safe-guards even with bbolt tool. Allocating 1TBs of memory becouse it consider random garbage from meta-page as block-size is not good and risk --> especially if diagnostic needs to be performed within production environment in order to avoid moving customer's data. 
If this would become a real problem -> we would need a 'force' flag to ignore such safe-guards.